internal/infrastructure/kubernetes/proxy/resource.go

```diff
@@ -402,7 +402,7 @@ func expectedContainerEnv(containerSpec *egv1a1.KubernetesContainerSpec, gateway
 			ValueFrom: &corev1.EnvVarSource{
 				FieldRef: &corev1.ObjectFieldSelector{
 					APIVersion: "v1",
-					FieldPath:  fmt.Sprintf("metadata.labels['%s']", corev1.LabelTopologyZone),
+					FieldPath:  fmt.Sprintf("metadata.annotations['%s']", corev1.LabelTopologyZone),
```

Comment:
why change from labels to annotations?

Reply:
Annotations from `Binding` will get copied to the pod unlike labels so it saves an extra resource to handle during webhook execution.

Reply:
ah cool !